mono/metadata/mono-perfcounters.c

```diff
@@ -1304,52 +1307,51 @@ custom_get_impl (SharedCategory *cat, MonoString *counter, MonoString* instance,
 }
 
 static const CategoryDesc*
-find_category (MonoString *category)
+find_category (const gunichar2* category, gint32 category_length)
 {
 	int i;
+	const CategoryDesc* desc = predef_categories;
 	for (i = 0; i < NUM_CATEGORIES; ++i) {
-		if (mono_string_compare_ascii (category, predef_categories [i].name) == 0)
-			return &predef_categories [i];
+		if (perfcounter_utf16_equal_ascii (category, category_length, desc->name, desc->name_length))
```

Comment:
Let's just keep the `predef_categories` access, it's much easier to read and understand than a hand-rolled iterator